app/controllers/hosts_controller.rb

```diff
@@ -681,8 +681,11 @@ def process_taxonomy
   # we don't need any has_many relation to determine what proxies are used and the view
   # renders only resulting templates set so the rest of form is unaffected
   def template_used
-    host = params[:id] ? Host.readonly.find(params[:id]) : Host.new
-    host.attributes = host_params.select { |k,v| !k.end_with?('_ids') }.except(:host_parameters_attributes)
+    host = params[:id] ? Host::Base.readonly.find(params[:id]) : Host.new
+    host.attributes = host_params.select do |k,v|
```

Comment:
This looks scary as well. Let's extract this to a method, shall we? Maybe a method that prepares "host".